library/platform_util.c

```diff
@@ -319,6 +319,18 @@ uint32_t mbedtls_platform_random_uint32( void )
               mbedtls_platform_random_uint16() );
 }
 
+void mbedtls_platform_random_buf( uint8_t *buf, size_t len )
+{
+    uint32_t i;
+
+    for( i = 0; i < len; i++ )
+    {
+       buf[i] = (uint8_t)mbedtls_platform_random_uint16();
```

Comment:
This will place an uint16_t in each of the buffer places that can hold uint8_t. It's not efficient.

Reply:
@AndrzejKurek , you are right. I tried to implement it using different way and use all received bytes  using following implementation :
```
void mbedtls_platform_random_buf( uint8_t *buf, size_t buf_size)
{
    uint32_t i;
    uint16_t *buf_uint16 = (uint16_t*)buf;
    size_t buf_size16 = buf_size/2;
    
    for(i = 0; i< buf_size16; i++)
    {
        buf_uint16[i] = mbedtls_platform_random_uint16();
    }
    if (buf_size%2 != 0)
    {
        buf[buf_size-1] = (uint8_t)mbedtls_platform_random_uint16();
    }
    return;
}
```
But I had some problems:
1.  `uint16_t *buf_uint16 = (uint16_t*)buf;` casting is problematic. It can cause alignment problem in some cases
2. The runtime of this implementation was worse then the existing one on L0 boards . It seems that / and % operations are heavy for them. 
3. It increased code size in 20 bytes

If you have any ideas how it can be improved, I will be happy to try

Reply:
try this:
```
void fill_buff(uint8_t* buf, size_t len)
{
    while(len > 1)
    {
        uint16_t val = get_uint16_t();
        buf[len-1] = val & 0xFF;
        buf[len-2] = val>>8;
        len -= 2;
    }
    if(len == 1)
    {
        buf[0] = get_uint16_t() & 0xFF;
    }
}
```

Reply:
@AndrzejKurek , tnx!
Replaced it with similar to your proposal implementation. It still increase code size in 20 bytes, but it seems to be more reasonable from random usage point of view. The run time size is the same as in my first implementation but I think it will be more significant with big buffers (14 bytes buffers were tested)